CaImAn/caiman_main.py

```diff
@@ -0,0 +1,316 @@
+
+    file_ = h5py.File(caiman_parameters.paths[defs.Parameters.Path.FILEPATH], 'r')
+    if defs.DoricFile.Dataset.IMAGE_STACK in file_[caiman_parameters.paths[defs.Parameters.Path.H5PATH]]:
+        images = np.array(file_[f"{caiman_parameters.paths[defs.Parameters.Path.H5PATH]}/{defs.DoricFile.Dataset.IMAGE_STACK}"])
+    else:
+        images = np.array(file_[f"{caiman_parameters.paths[defs.Parameters.Path.H5PATH]}/{defs.DoricFile.Deprecated.Dataset.IMAGES_STACK}"])
+
+    logging.debug(images.shape)
+
+    print(cm_defs.Messages.WRITE_IMAGE_TIFF, flush=True)
+    imwrite(caiman_parameters.params_caiman["fnames"], images.transpose(2, 0, 1))
+    del images
+
+    opts = params.CNMFParams(params_dict=caiman_parameters.params_caiman)
+    opts_dict, advanced_settings = set_advanced_parameters(opts, caiman_parameters.advanced_settings)
+
+    #Update parameters and Advanced Setting
+    opts.change_params(opts_dict)
+    caiman_parameters.parameters[defs.Parameters.danse.ADVANCED_SETTINGS] = advanced_settings.copy()
+
+    if bool(caiman_parameters.parameters[defs.Parameters.danse.CORRECT_MOTION]):
+        # MOTION CORRECTION
+        print(cm_defs.Messages.MOTION_CORREC,  flush=True)
+        # do motion correction rigid
+        try:
+            mc = MotionCorrect(caiman_parameters.params_caiman["fnames"], dview=dview, **opts.get_group("motion"))
+        except TypeError:
+            utils.print_to_intercept(cm_defs.Messages.PARAM_WRONG_TYPE)
+            sys.exit()
+        except Exception as error:
+            utils.print_error(error, cm_defs.Messages.MOTION_CORREC)
+            sys.exit()
+
+        mc.motion_correct(save_movie=True)
+        fname_mc = mc.fname_tot_els if caiman_parameters.params_caiman["pw_rigid"] else mc.fname_tot_rig
+
+        bord_px = 0 if caiman_parameters.params_caiman["border_nan"] == "copy" else caiman_parameters.params_caiman["bord_px"]
+        fname_new = cm.save_memmap(fname_mc, base_name="memmap_", order='C', border_to_0=bord_px)
+
+    else:  # if no motion correction just memory map the file
+        fname_new = cm.save_memmap([caiman_parameters.params_caiman["fnames"]], base_name="memmap_", order='C', border_to_0=0)
+
+
+    # load memory mappable file
+    Yr, dims, T = cm.load_memmap(fname_new)
+    images = Yr.T.reshape((T,) + dims, order='F')
+
+    try:
+        print(cm_defs.Messages.START_CNMF, flush=True)
+        cnm = cnmf.CNMF(n_processes = n_processes, dview=dview, params=opts)
+        print(cm_defs.Messages.FITTING, flush=True)
+        cnm.fit(images)
+
+        print(cm_defs.Messages.EVA_COMPO, flush=True)
+        cnm.estimates.evaluate_components(images, cnm.params, dview=dview)
+    except TypeError:
+        utils.print_to_intercept(cm_defs.Messages.PARAM_WRONG_TYPE)
+        sys.exit()
+    except Exception as error:
+        utils.print_error(error, cm_defs.Messages.START_CNMF)
+        sys.exit()
+
+    ### Save results to doric file ###
```

Comment:
I think I edited a bit this part for minian, make sure it looks similar, including the comments